tests/test_examples.py

```diff
@@ -1336,30 +1337,49 @@ def test_second_stage_optimization_CoilSet():
     np.testing.assert_allclose(field[0].current, 0, atol=1e-12)
 
 
-# TODO: replace this with the solution to Issue #1021
 @pytest.mark.unit
-def test_optimize_with_fourier_planar_coil():
-    """Test optimizing a FourierPlanarCoil."""
+def test_optimize_with_all_coil_types(DummyCoilSet, DummyMixedCoilSet):
+    """Test optimizing for every type of coil and dummy coil sets."""
+    sym_coils = load(load_from=str(DummyCoilSet["output_path_sym"]), file_format="hdf5")
+    asym_coils = load(
+        load_from=str(DummyCoilSet["output_path_asym"]), file_format="hdf5"
+    )
+    mixed_coils = load(
+        load_from=str(DummyMixedCoilSet["output_path"]), file_format="hdf5"
+    )
+    nested_coils = MixedCoilSet(sym_coils, mixed_coils)
+
+    def test(c, method):
+        target = 11
+        obj = ObjectiveFunction(
+            (
+                CoilLength(c, target=target),
+                QuadraticFlux(eq=Equilibrium(), field=c, vacuum=True),
```

Comment:
Instead of instantiating an `Equilibrium` each time test is called, just pass it in to `test` as an argument or something like that, to avoid the cost of making a new eq each time.